packages/devtools_app/lib/src/utils.dart

```diff
@@ -1069,3 +1069,54 @@ String prettyTimestamp(
   );
   return DateFormat.Hms().format(timestampDT); // HH:mm:ss
 }
+
+/// A [ChangeNotifier] that holds a list of data.
+///
+/// When [value] is replaced with something that is not equal to the old
+/// value as evaluated by the equality operator ==, this class notifies its
+/// listeners.
+///
+/// This class also exposes methods `addAndNotify` and `addAllAndNotify` that
+/// will notify listeners upon the list [value] being modified.
+class ListValueNotifier<T extends List<V>, V> extends ChangeNotifier
```

Comment:
I would recommend removing T. and list make the value be List<V>

Reply:
restructured PTAL